### PR TITLE
fixing code

### PR DESCRIPTION
**Pull request by ShadiMah**

### CODE REVIEW

File: main_adapter.py [hunk 1]
```diff
@@ -12,6 +12,12 @@ class ModelAdapter(dl.BaseModelAdapter):
     def __init__(self, model_entity: dl.Model):
         if os.environ.get("NVIDIA_NIM_API_KEY", None) is None:
             raise ValueError(f"Missing API key: NVIDIA_NIM_API_KEY")
+        self.api_key = os.environ.get("NVIDIA_NIM_API_KEY", None)
```
Comment (datafranch) [inline main_adapter.py:15]: I always question myself about this: is it better to get the api key just once at service initialization or before every request we do?

File: main_adapter.py [hunk 2]
```diff
@@ -93,14 +129,15 @@ def predict(self, batch, **kwargs):
                 if len(system_prompt) > 0:
                     messages.append({"role": "system",
                                      "content": system_prompt})
-                messages.extend([{"role": "assistant",
-                                  "content": context},
-                                 {"role": "user",
-                                  "content": question}])
+                messages.extend([
+                    {"role": "user",
```
Comment (datafranch) [inline main_adapter.py:133]: this is what we discussed about the assistant role, right?

**Review by datafranch — APPROVED** [review]

File: main_adapter.py (hunk 1 above)
Comment (ShadiMah) [reply main_adapter.py:15]: here it is used in multiple places, so to avoid the code duplication i moved it to init

File: main_adapter.py (hunk 2 above)
Comment (ShadiMah) [reply main_adapter.py:133]: it is not 100% that, 
this is how the nvidia models work, there will be a different refactor

Comment (RoniDataloopAi) [reply main_adapter.py:133]: in call_coding_model you don't use stream?

**Pull request merged**